library/network_connections.py

```diff
@@ -993,16 +995,19 @@ def connection_create(self, connections, idx, connection_current=None):
                     NM.SETTING_IP_CONFIG_ROUTE_METRIC, ip["route_metric4"]
                 )
             for d in ip["dns"]:
-                if d["family"] == socket.AF_INET:
-                    s_ip4.add_dns(d["address"])
-            for s in ip["dns_search"]:
-                s_ip4.add_dns_search(s)
-            s_ip4.clear_dns_options(True)
-            for s in ip["dns_options"]:
-                s_ip4.add_dns_option(s)
-
+                if not ipv4_is_disabled:
+                    if d["family"] == socket.AF_INET:
```

Comment:
This will silently ignore IPv4 DNS settings if IPv4 is disabled but this needs to create a failure.

Reply:
make sense, thanks

Reply:
Code updated.
I suddenly realized that we should raise errors when `dns_search` (or `dns_options`) is defined but dns nameserver is not defined. Right?

Reply:
+1

Reply:
@liangwen12year Any update on this?

Reply:
I think the correct  validation is:
- raise errors when ipv6 dns option is defined and ipv6 is disabled
- raise errors when `dns_search` (or `dns_option`) is defined but both ipv6 and ipv4 is disabled.

Bear in mind that we do not need to clearly define nameserver, we can always get nameserver via DHCP as long as corresponding ip method is not disabled. Correct me if I was wrong , @thom311 ?